### ISSUE TITLE
Login field does not trim trailing spaces on email

### ISSUE THREAD

**Issue by abhchand** [hay-kot/mealie#475]
**Describe the bug**

If the user logs in with an email that has trailing spaces the login will fail.

Many sites trim whitespaces from email on login, since email can often be pasted or filled in from elsewhere. 

**Steps To Reproduce**

1. Click "Login"
2. Type in `"changeme@email.com   "` (without quotes)
3. Type in a valid password (e.g. `MyPassword`) 
4. Try logging in - note the error message
5. Try removing the trailing spaces in the email and logging in again, it will succeed 

**Expected behavior**

There's 2 options here, depending on which product behavior you prefer

1. Automatically trim whitespace when logging in
2. Follow the lead of sites like digitalocean.com and mark the email as `invalid` immediately so the user is aware of whitespace (see screenshot)

**Actual Behavior**

User can not log in with trailing whitespace on email

**Screenshots**

<img width="590" alt="Screen Shot 2021-06-09 at 11 55 18 AM" src="https://user-images.githubusercontent.com/13787645/121389569-8d300800-c91a-11eb-81d5-e7352b045bb7.png">

How digitalocean handles it:

<img width="535" alt="Screen Shot 2021-06-09 at 11 57 49 AM" src="https://user-images.githubusercontent.com/13787645/121389570-8dc89e80-c91a-11eb-98ff-bf7431383a49.png">

**Device Information (please complete the following information):**
 - OS: Ubuntu 20.04 LTS
 - Deployment: `docker run` with `hkotel/mealie:latest`
 - Browser: Firefox 89.0

**Comment by zierbeek** [comment]
I think this is fixed in the dev branch (release 0.5.0). Tested it and seems fixed. Correct me if I'm wrong @hay-kot

**Comment by hay-kot** [comment]
> I think this is fixed in the dev branch (release 0.5.0). Tested it and seems fixed. Correct me if I'm wrong @hay-kot

I don't see anything that explicitly would do that. There is already a validator in the mixins for whitespace though so that may be an easy option.

**Comment by zierbeek** [comment]
I can not replicate in the demo neither in my piece

**Comment by hay-kot** [comment]
> I think this is fixed in the dev branch (release 0.5.0). Tested it and seems fixed. Correct me if I'm wrong @hay-kot

I was wrong. I did in-fact implement the trimming of whitespace on usernames/emails for login

```python
# mealie/schema/auth.py
...
class TokenData(BaseModel):
    username: Optional[constr(to_lower=True, strip_whitespace=True)] = None
```

Will mark as fixed pending release

**Comment by abhchand** [comment]
@hay-kot - thanks!

And a huge thanks to everyone that contributes here. Really appreciate the time you put in here, Mealie has been quite useful.
